collect_app/src/main/java/org/odk/collect/android/application/Collect.java

```diff
@@ -129,15 +121,21 @@ protected void attachBaseContext(Context base) {
     @Override
     public void onCreate() {
         super.onCreate();
-        ExternalFilesUtils.testExternalFilesAccess(this);
-
         singleton = this;
-        setupDagger();
-        DaggerUtils.getComponent(this).inject(this);
 
-        applicationInitializer.initialize();
-        fixGoogleBug154855417();
-        setupStrictMode();
+        CrashHandler crashHandler = CrashHandler.install(this);
+        boolean shouldLaunch = crashHandler.checkConditions(() -> {
+            ExternalFilesUtils.testExternalFilesAccess(this);
```

Comment:
We might want to put other checks in here to prevent other crashes from happening down the road.